### DISCUSSION TITLE
How to redirect signed in users?

### DISCUSSION THREAD

**Discussion by AugustinSorel** [nextauthjs/next-auth#8910] [Help]
### Summary

Hi everyone,

I am trying to protect my `/sign-in` page from signed in user by redirecting them to another page. However I have been stuck with this problem for a while now. I have found [this stack overflow discussion](https://stackoverflow.com/questions/76175812/prevent-authenticated-users-to-access-custom-sign-in-page-with-next-auth-middlew), however both answers provided do not work.

Here is what my current middleware looks like.
All it does right now is protecting my private pages to non signed users.

How could I extend this middleware to redirect users to a specific page if they are signed in ?

```ts
export { default } from "next-auth/middleware";

export const config = { matcher: ["/dashboard/:path*", "/exercises/:path*"] };
```

### Additional information

To clarify here is what I am trying to do.

I am trying to protect the routes :

* `/dashboard` `/exercises/*` - Those two routes should only be accessible by signed in users
* `/` `/sign-in` `/verify-request` - Those three routes should only be accessible by not signed in users

### Example

https://github.com/augustinsorel/gym-graphs

**Comment by vnevermore** [comment]
hello @AugustinSorel , did you manage to solve this? i am looking for the exact same thing

**Reply by AugustinSorel** [reply]
Unfortunatly I did not solve it with a middleware. But instead I am manually redirecting the user like so:

```ts
import { getServerSession } from "next-auth";
import { authOptions } from "@/lib/auth";

const Page = async () => {
  const session = await getServerSession(authOptions);

  if (!session?.user.id) {
    return redirect("/dashboard");
  }

  return ...
};
```

**Reply by vnevermore** [reply]
oh well thanks for your reply, i got it like this aswell for now but i hate the idea of getting servers session on page server render and then redirect while i could do that on the middleware level cleaner and in a more efficient way. Anyway if i solve it somehow i ll let you know. Have a great day
